clippy_lints/src/eta_reduction.rs

```diff
@@ -174,9 +174,9 @@ fn match_types(lhs: &ty::TyKind<'_>, rhs: &ty::TyKind<'_>) -> bool {
 
 fn get_type_name(cx: &LateContext<'_, '_>, kind: &ty::TyKind<'_>) -> String {
```

Comment:
The problem is this function should be taking `ty: Ty`.
`TyKind` should really never be mentioned by code outside of `rustc::ty`.